cou/steps/__init__.py

```diff
@@ -134,6 +134,17 @@ def __repr__(self) -> str:
         """
         return f"UpgradeStep({self.description})"
 
+    def __bool__(self) -> bool:
+        """Boolean magic method for UpgradeStep.
+
+        :return: True if there is at least one coroutine in a UpgradeStep
+        or in its sub steps.
+        :rtype: bool
+        """
+        if self._coro or any(step._coro for step in self.sub_steps):
+            return True
+        return any(bool(step) for step in self.sub_steps)
```

Comment:
I think that logic will be same if you use `and`, or not?
```suggestion
        return self._coro and any(bool(step) for step in self.sub_steps):
```

Reply:
No. The Top plan doesn't have a coroutine (self._coro) but their sub steps will have.

Reply:
That is true, but `return self._coro is not None or any(bool(step) for step in self.sub_steps)` will work.

Reply:
That is what I did in the next commit